Sources/DeploymentTargetAWSLambdaCommon/LambdaCommon.swift

```diff
@@ -70,3 +71,28 @@ public extension AnyOption where OuterNS == DeploymentOptionsNamespace {
         ResolvedOption(key: .lambdaDescription, value: value)
     }
 }
+
+public struct LambdaDeployedSystem: AnyDeployedSystem {
+    public var userInfo: Data
+    
+    public var nodes: Set<DeployedSystemNode>
+    
+    public var deploymentProviderId: DeploymentProviderID
+    
+    public var openApiDocument: OpenAPI.Document
+    
+    public init<T: Encodable>(
+        deploymentProviderId: DeploymentProviderID,
+        nodes: Set<DeployedSystemNode>,
+        userInfo: T?,
+        userInfoType: T.Type = T.self,
```

Comment:
Would it be possible to make the `AnyDeployedSystem` have an associated type that is Codable so you don't have to implement the custom encoding there? E.g. a default implementation of `AnyDeployedSystem` just decodes the associated type dynamically.